lib/client/connect.js

```diff
@@ -1,19 +1,59 @@
 KadiraDebug = window.KadiraDebug = {};
 KadiraInfo = new Mongo.Collection('kdInfo');
 
-var debug = getQueryParamFromURI("kadira_debug");
-var accessToken = (debug)? (getQueryParamFromURI("access_token") || null) : null;
-Meteor.subscribe('kadira.debug.client.auth', accessToken);
+var env = Utils.getAppEnv();
 
-Meteor.subscribe('kadira.debug.client.listeners');
+var debug = getQueryParamFromURI('kadira_debug');
+var accessToken = (debug)? (getQueryParamFromURI('access_token') || null) : null;
+
+if(env === 'production' && !accessToken) {
+  throw new Meteor.Error('401', 'Unauthorized.');
+}
+
+Meteor.subscribe('kadira.debug.client.auth', accessToken, {
+  onReady: function () { 
+    // create a unique client for this client
+    // works accross hot code reloads as well
+    var clientId = Session.get('kdClientId');
+    if(!clientId) {
+      clientId = Random.id(8);
+      Session.set('kdClientId', clientId);
+    }
+    KadiraDebug.clientId = clientId;
+
+    // create a unique Id for the browser
+    var browserId = Meteor._localStorage.getItem('kdBrowserId');
+    if(!browserId) {
+      function setName(err, name) {
+        if(err) {
+          browserId = Random.id(8);
+        } else {
+          browserId = name;
+        }
+        Meteor._localStorage.setItem('kdBrowserId', browserId);
+        // invoke the starting process
+        KadiraDebug._init(browserId, clientId);
+      }
+      
+      Meteor.call('kadira.debug.client.getBrowserName', navigator.userAgent, setName);
+    } else {
+      // invoke the starting process
+      KadiraDebug._init(browserId, clientId);
+    }
+  },
+  onError: function () { 
```

Comment:
This should be onStop with `err` as the first arguments.
if `err` is exists that's an error. If not, that's some other disconnection.